### PR TITLE
Add "ease" animation tokens

### PR DESCRIPTION
**Pull request by LauraAubin**
Part of https://github.com/Shopify/polaris-tokens/issues/79

### What is this PR doing?

<sup>*Note that this example now says "ease" instead.</sup>
<kbd>gif</kbd>

Screen Shot 2019-12-10 at 7 05 27 PM

This adds the `ease` tokens which have 3 variations:
- default
- appear
- disappear

This token is generated from an array of 4 numbers which makes up the css value. This way we can isolate the value from the css so devices that don't support `cubic-bezier` can still make use of this token in an alternate way.

### How to 🎩

Run `yarn dist`, the docs html should still have a section for `ease` and no other files should be modified.

### CODE REVIEW

**Comment by johanstromqvist** [comment]
Sorry to be late to the game, two quick thoughts.

@alex-page had a suggestion and some reasoning for going with `ease` instead of `easing`, e.g. `ease-default`.

I'm not 100% sure about the naming convention yet, `default`, `appear`, and `disappear`. It's descriptive but unconventional. The descriptive name will make it easier to choose the right ease in most cases, but on some occasion it won't. Ideally we could reference a timing function with either the conventional or the descriptive name.

E.g:
’ease-in = cubic-bezier(0.5, 0.1, 1, 1)’
’ease-disappear = ease-in’

**Comment by johanstromqvist** [comment]
Regarding the "Usage" column. Both easing in general and the different variants would need more extensive documentation, but here are some concise descriptions that might work in a cheatsheet or overview.

`Ease: cubic-bezier(0.4, 0.22, 0.28, 1)` (Default)
- Has both acceleration and break effect
- Useful when an element's before and after states are both visible within screen boundaries
- Useful when an exiting or entering element should be perceived as hidden and not gone, e.g. when closing a menu or sheet
- Useful when synchronizing the transitions of multiple properties or elements
- Can safely be used to any transition, although they are not optimal
- Extreme short version: "Use by default"

`Ease-in: cubic-bezier(0.5, 0.1, 1, 1)` (Disappear)
- Has acceleration, but no break effect
- Useful when elements are disappearing
- Extreme short version: "For disappearing elements"

`Ease-out: cubic-bezier(0, 0, 0.13, 1)` (Appear)
- Has break effect, but no acceleration
- Useful when elements are appearing
- Extreme short version: "For appearing elements"

`Linear: cubic-bezier(0, 0, 1, 1)` (Constant)
- Has constant speed, no acceleration or break effect
- Useful only when speed variations are undesired
- Extreme short version: "For intentionally constant speed only"

**Comment by LauraAubin** [comment]
@johanstromqvist These have been renamed to `ease`, and are following your initial naming convention.

In regards to your second point [here](https://github.com/Shopify/polaris-tokens/pull/87#issuecomment-564659860), since we're not 100% sure about the naming convention, I don't think it makes sense to also include the descriptive name mapping at this time.

I'd like to add these tokens to the system so we can start using them and see if @sarahill can include them in her user studies.

File: tokens/ease.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+props:
+  - name: ease-default
+    value: [0.4, 0.22, 0.28, 1]
+  - name: ease-appear
+    value: [0, 0, 0.13, 1]
+  - name: ease-disappear
+    value: [0.5, 0.1, 1, 1]
+global:
+  type: ease
```
Comment (kaelig) [inline tokens/ease.yml]:
```suggestion
  type: easing
```

File: gulpfile.js [hunk 2]
```diff
@@ -20,7 +20,22 @@ theo.registerValueTransform(
   (prop) => (prop.get('value') === 0 ? 0 : ms(prop.get('value'))),
 );
 
-theo.registerTransform('web/js', ['color/rgb', 'timing/ms-unitless']);
+theo.registerValueTransform(
+  'ease/web',
+  (prop) => prop.get('type') === 'ease',
```
Comment (kaelig) [inline gulpfile.js]:
```suggestion
  (prop) => prop.get('type') === 'easing',
```

File: formats/docs.html.js [hunk 3]
```diff
@@ -350,6 +364,7 @@ class Styleguide {
               ${this.renderSection('text-color', 'Text Colors')}
               ${this.renderSection('text-shadow', 'Text Shadows')}
               ${this.renderSection('time', 'Time')}
+              ${this.renderSection('cubic-bezier', 'ease')}
```
Comment (kaelig) [inline formats/docs.html.js]:
```suggestion
              ${this.renderSection('cubic-bezier', 'Easing')}
```

File: tokens/index.yml [hunk 4]
```diff
@@ -3,6 +3,7 @@
 imports:
   - ./colors.yml
   - ./duration.yml
+  - ./ease.yml
```
Comment (kaelig) [inline tokens/index.yml]:
```suggestion
  - ./easing.yml
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,7 +5,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-<!-- ## [Unreleased] -->
+## [Unreleased]
+
+- Added `ease` tokens ([#87](https://github.com/Shopify/polaris-tokens/pull/87))
```
Comment (kaelig) [inline CHANGELOG.md]:
```suggestion
- Added `easing` tokens ([#87](https://github.com/Shopify/polaris-tokens/pull/87))
```

File: tokens/ease.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+props:
+  - name: ease-default
+    value: [0.4, 0.22, 0.28, 1]
```
Comment (kaelig) [inline tokens/ease.yml]: It'd be nice to have usage guidelines for each one (in a `comment` field). Note that we'll be enforcing such documentation in the future.

Comment (LauraAubin) [reply tokens/ease.yml]: Will this documentation follow some sort of structure? Similar to how readme files in Polaris are used to generate the style guide?

Comment (kaelig) [reply tokens/ease.yml]: We'll probably have something like Markdown to support complex usage guidelines, like in https://polaris.shopify.com/design/colors#section-color-usage – for now I think one-liners that go straight to the point would be great, like we do for icons in polaris-icons.shopify.com

**Comment by johanstromqvist** [comment]
> @johanstromqvist These have been renamed to `ease`, and are following your initial naming convention.
> 
> In regards to your second point [here](https://github.com/Shopify/polaris-tokens/pull/87#issuecomment-564659860), since we're not 100% sure about the naming convention, I don't think it makes sense to also include the descriptive name mapping at this time.
> 
> I'd like to add these tokens to the system so we can start using them and see if @sarahill can include them in her user studies.
> 
> #### Update Jan 9
> * Sara completed her round of user testing with the color names and already synced with Johan on a naming strategy.
> * Naming convention is currently being looked into.

After seeing that [IBM design](https://www.ibm.com/design/language/motion-ui/basics/) calls their easing curves `standard`, `exit` and `enter`, I'm more confident that a descriptive and instructional naming convention is the better choice. My assumption is that this will catch 95%+ of the use cases, and if that's true, I think it should be considered successful. We simply can't aim for 100% use cases at system level.

The conventions I'm considering are:
- default, appear, disappear
- stay, add, remove
- natural, accelerate, brake

These would have some different pros and cons, and if we are testing naming conventions it would be meaningful to learn what designers find the most intuitive. CC @sarahill

**Comment by alex-page** [comment]
Closing this issue for now. This will be very important in the near future. I would love to re-open it or start a conversation with everyone here when we expand `polaris-tokens`.

**Pull request closed without merging**